### PR TITLE
Add brief mention of MISRA and HICPP to justify valueless after move

### LINKED ISSUE #381: Add explicit discussion of why valueless_after_move is needed
Needed for https://isocpp.org/files/papers/P3124R0.html#poll-1-send-p3019r6-vocabulary-types-for-composite-class-design-to-library-working-group-for-c26

### PR DESCRIPTION
**Pull request by jbcoe**
closes #381

### CODE REVIEW

File: DRAFT.md [hunk 1]
```diff
@@ -489,6 +491,15 @@ With some suitably compelling motivation, the `explicit` keyword could be
 removed from some constructors in a later revision of the C++ standard without
 rendering code ill-formed.
 
+### The `valueless_after_move` member function
+
+Both `indirect` and `polymorphic` have a valueless state that is used to query
+the object state. By intention, this member function should not be often called:
```
Comment (nbx8) [inline DRAFT.md:497]: Intentionally, this member function should not often be called:

File: DRAFT.md [hunk 2]
```diff
@@ -489,6 +491,15 @@ With some suitably compelling motivation, the `explicit` keyword could be
...
+it should be clear through static analysis whether or not an object has been
+moved from. The member function to allows explicit checks for the valueless
```
Comment (nbx8) [inline DRAFT.md:499]: The member function allows explicit checks...

File: DRAFT.md [hunk 3]
```diff
@@ -1479,6 +1490,12 @@ _A polymorphic value-type for C++_, J. B. Coe, S. Parent 2019
 _A Free-Store-Allocated Value Type for C++_, J. B. Coe, A. Peacock 2022
 [https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p1950r2.html]
 
+_MISRA Language Guidelines
+[https://ldra.com/misra/]_
```
Comment (nbx8) [inline DRAFT.md:1494]: _MISRA Language Guidelines_
[[https://ldra.com/misra/]

**Review by nbx8 — DISMISSED** [review]
Minor edits suggested, otherwise good to go.

File: DRAFT.md (hunk 1 above)
Comment (jbcoe) [reply DRAFT.md]: Dropped "Intentionally" altogether - it implies other elements of our design may be accidents :-)

**Pull request merged**